sky/packages/sky/lib/src/rendering/custom_layout.dart

```diff
@@ -16,41 +18,45 @@ abstract class MultiChildLayoutDelegate {
   /// can reflect that.
   Size getSize(BoxConstraints constraints) => constraints.biggest;
 
+  /// True if a non-null LayoutChild was provided for the specified id.
+  bool isChild(Object childId) => _idToChild[childId] != null;
+
   /// Ask the child to update its layout within the limits specified by
   /// the constraints parameter. The child's size is returned.
-  Size layoutChild(int childIndex, BoxConstraints constraints) {
-    final RenderBox child = _indexToChild[childIndex];
+  Size layoutChild(Object childId, BoxConstraints constraints) {
+    final RenderBox child = _idToChild[childId];
```

Comment:
Should this return Size.zero if !hasChild(childId) ?  That might simplify the callers.